lux/lib/lux/prisms/discord/command_prism.ex

```diff
@@ -0,0 +1,425 @@
+  @timeout :timer.seconds(10)
+  @max_retries 3
+  @retry_delay 1000 # 1 second
+
+  # Message types
+  @message_types %{
+    pong: 1,
+    channel_message: 4,
+    deferred_channel_message: 5,
+    deferred_update_message: 6,
+    update_message: 7,
+    autocomplete_result: 8
+  }
+
+  # Component types
+  @component_types %{
+    action_row: 1,
+    button: 2,
+    string_select: 3,
+    text_input: 4,
+    user_select: 5,
+    role_select: 6,
+    mentionable_select: 7,
+    channel_select: 8
+  }
+
+  # Message flags
+  @flags %{
+    ephemeral: 64
+  }
+
+  @type interaction_response :: {:ok, map()} | {:error, map()}
+  @type request_params :: %{
+    method: :get | :post | :put | :patch | :delete,
+    url: String.t(),
+    body: map() | nil
+  }
+  @type validation_result :: {:ok, map()} | {:error, map()}
+
+  @doc """
+  Adds Discord bot token to request headers.
+  """
+  def add_auth_header(prism) do
+    token = Application.fetch_env!(:lux, :discord_token)
+    %{prism | headers: prism.headers ++ [{"Authorization", "Bot #{token}"}]}
+  end
+
+  @doc """
+  Handles the Discord API request with proper error handling and timeout management.
+  """
+  @impl true
+  def handler(input, ctx) do
+    with {:ok, request} <- prepare_request(input),
+         {:ok, response} <- execute_request(request, @max_retries) do
+      format_response(response)
+    end
+  end
+
+  @doc """
+  Prepares the API request based on validated parameters.
+  """
+  @spec prepare_request(map()) :: {:ok, request_params()} | {:error, map()}
+  def prepare_request(params) do
+    try do
+      request = build_request(params)
+      {:ok, request}
+    rescue
+      e in RuntimeError ->
+        Logger.error("Error preparing request: #{inspect(e)}")
+        {:error, %{type: "preparation_error", message: e.message}}
+    end
+  end
+
+  @doc """
+  Executes the API request asynchronously with timeout handling and retries.
+  """
+  @spec execute_request(request_params(), non_neg_integer()) :: interaction_response()
+  def execute_request(request, retries_left) do
+    task = Task.async(fn ->
+      try do
+        make_request(request)
+      rescue
+        e in Req.Error ->
+          {:error, %{type: "request_error", message: e.message}}
+      end
+    end)
+
+    case Task.yield(task, @timeout) do
+      {:ok, {:error, %{type: "rate_limit"} = error}} when retries_left > 0 ->
+        Logger.warn("Rate limited by Discord API, retrying after #{error.retry_after} seconds")
+        Process.sleep(error.retry_after * 1000)
+        execute_request(request, retries_left - 1)
+
+      {:ok, result} -> result
+
+      nil ->
+        Task.shutdown(task)
+        Logger.error("Request timed out")
+        {:error, %{type: "timeout", message: "Request timed out"}}
+    end
+  end
+
+  # Private functions
+
+  @spec make_request(request_params()) :: interaction_response()
+  defp make_request(%{method: method, url: url, body: body}) do
```

Comment:
Left some comments on another PR about this. Perhas we could extract this request code somewhere and reuse it inside prisms.